vllm/worker/hpu_model_runner.py

```diff
@@ -1142,29 +1157,37 @@ def _prepare_decode(
         block_groups = padding_fn(block_groups, -1)
         block_usage = padding_fn(block_usage, 1)
 
-        block_list = torch.tensor(block_list,
-                                  dtype=torch.int,
-                                  device=self.device)
+        block_list = torch.tensor(block_list, dtype=torch.int, device='cpu')
         block_groups = torch.tensor(block_groups,
                                     dtype=torch.int,
-                                    device=self.device)
+                                    device='cpu')
         block_usage = torch.tensor(block_usage,
                                    dtype=self.model_config.dtype,
-                                   device=self.device)
+                                   device='cpu')
         slot_mapping = torch.tensor(slot_mapping,
                                     dtype=torch.long,
-                                    device=self.device)
-
-        block_indices, block_offsets = precompute_indices_and_offsets(
-            self.block_size, slot_mapping, False)
+                                    device='cpu')
+
+        input_tokens = input_tokens.to(  # type: ignore
+            self.device, non_blocking=True)
+        input_positions = input_positions.to(  # type: ignore
+            self.device, non_blocking=True)
+        block_list = block_list.to(  # type: ignore
+            self.device, non_blocking=True)
+        block_groups = block_groups.to(  # type: ignore
+            self.device, non_blocking=True)
+        block_usage = block_usage.to(  # type: ignore
+            self.device, non_blocking=True)
+        slot_mapping = slot_mapping.to(  # type: ignore
```

Comment:
at which step this transfer is synchronized? make_metadata does that?

Reply:
also shouldnt source be in pinned memory for device side to be non blocking as per https://pytorch.org/tutorials/intermediate/pinmem_nonblock.html?

Reply:
Regarding synchronization - bridge takes care of synchronizing it. 
As for pinning - does this change anything on HPU?

Reply:
@michalkuligowski Pinning memory is not supported on HPU [- we don't do that in current vllm implementation, eg. ](https://github.com/HabanaAI/vllm-fork/blob/fb308c9b80c3467bbb71db123b217252e006aa8b/vllm/utils.py#L756C1-L756C66) Even when I tried to switch on memory pinning manually, it didn't work.